DOCS-8417-cluster-encrypt-v4.3-cv (#1903)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/mule-high-availability-ha-clusters.adoc

You can also choose to define the minimum number of machines that a cluster requires to remain in an operational state. This configuration grants you a consistency improvement in the overall performance.

== Encrypt Communication Between Cluster Nodes

> You can encrypt traffic between cluster nodes by defining the `mule.cluster.network.encryption.key` property. You can define this property in the `wrapper.conf` file, or as a system property.

Reviewer: Perfectly fine as-is.
Another way:
```suggestion
You can encrypt traffic between cluster nodes by defining the `mule.cluster.network.encryption.key` property, either in the `wrapper.conf` file or as a system property.
```

When you define an encryption key, all communication between cluster nodes is encrypted using the defined key and the AES encryption algorithm.

[[cluster_fips]]
== Enable Clustering in FIPS Mode

To enable clustering in FIPS mode, complete the following tasks:

* Run Mule runtime engine in FIPS mode. +

> See xref:fips-140-2-compliance-support.adoc[FIPS Compliant support] for configuration instructions.

Reviewer: ```suggestion
See xref:fips-140-2-compliance-support.adoc[FIPS-Compliant Support] for configuration instructions.
```

* Encrypt communication between cluster nodes. +
Define the `mule.cluster.network.encryption.key` property, either in the `wrapper.conf` file or as a system property.

=== Warnings

> * If you don't define the encryption key, the application fails to deploy with message: `Cluster key must be defined in FIPS mode.`.

Reviewer: ```suggestion
* If you don't define the encryption key, the application fails to deploy with the message `Cluster key must be defined in FIPS mode.`.
```

> * Use the same encryption key for all the clustered nodes, otherwise this feature won't work.

Reviewer: parallelism
``suggestion
* If you don't use the same encryption key for all the clustered nodes, this feature won't work.
```

> * Running a clustered environment in FIPS mode causes a negative impact in performance, because all the communication between nodes will be encrypted (and decrypted).

Reviewer: ```suggestion
* Running a clustered environment in FIPS mode causes a negative impact in performance, because all the communication between nodes is encrypted (and decrypted).
```
